Add options to use system PCL and OpenCV

iamwolf opened the pull request:
Most of our build time is taken up by PCL and OpenCV in externals, where we are nowadays using mainline revisions. This PR adds the (disabled by default) option to use an external copy of PCL and OpenCV.

This has been tested on a fresh install of Ubuntu Trusty and works with

a) The Jochen Sprickerhof PPA for PCL, i.e.:
```
sudo add-apt-repository -y ppa:v-launchpad-jochen-sprickerhof-de/pcl
sudo apt-get update
sudo apt-get install -y libpcl-1.7-all
```
and to remove ``sudo apt-get remove libpcl-1.7-all``.

b) An in-house packaged version of OpenCV 2.4.12.3 that matches the one we have in externals:
```
wget http://terminator.robots.inf.ed.ac.uk/public/opencv_2.4.12.2.deb
sudo dpkg -i opencv_2.4.12.2.deb
```
and to remove: ``sudo dpkg -r opencv``

These options are disabled by default on purpose. I'd like to add them, however, to shave off time e.g. for build servers. After this PR, one can use the above mentioned versions :
```
cd software/externals
mkdir pod-build && cd pod-build
cmake .. -DUSE_SYSTEM_PCL:BOOL=ON -DUSE_SYSTEM_OPENCV:BOOL=ON
```

patmarion commented:
+1

patmarion commented on software/externals/cmake/externals.cmake:
```diff
@@ -307,6 +307,29 @@ if(NOT APPLE)
+  list(REMOVE_ITEM libmultisense_depends
+    opencv
+  )
+endif()
```
I guess doing list REMOVE_ITEM works fine, but did you see how I used the lcm_proj variable for the optional USE_SYSTEM_LCM?

iamwolf replied:
I've seen you use ${lcm_proj} which I reckon is empty if not defined - would that be a cleaner way for this? (Happy to amend)

patmarion replied:
Yeah I think I prefer using the variable (which is empty if not defined) because then you don't have to explicitly undo the dependencies.

iamwolf commented:
@patmarion Thanks for the feedback. I've addressed your comments, added a little note to the very end of the README and tested it with the most recent changes on a fresh Ubuntu installation. Also, I've added another commit changing the git URLs from SSH to HTTPS for public repositories (apologies for not separating this out as another Pull Request). Rationale behind it is same as for similar PRs - allow to clone and build on a machine that does not have a public key associated with a GitHub account (e.g. a fresh container/VM).

patmarion commented:
Looks great!  I'm not sure why we had the ssh:// convention, thanks for changing it!
